File: pinyougou-parent-module/pinyougou-manager-web/src/main/java/cn.itcast.web.controller/BrandController.java
package cn.itcast.web.controller;


import cn.itcast.pojo.TbBrand;
import cn.itcast.service.BrandService;
import com.alibaba.dubbo.config.annotation.Reference;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll.do")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }

    /**
     * 分页查询
     * @param currPage 当前页
     * @param pageSize 每页行数
     * @return
     */
    @RequestMapping("/findByPage.do")
    public PageResult<TbBrand> findByPage(Integer currPage ,Integer pageSize){
        return brandService.findByPage(currPage,pageSize);

    }


    /**
     * 添加
     * @param tbBrand 品牌实体
     * @return
     */
    @RequestMapping("/save.do")
    public Result save(@RequestBody TbBrand tbBrand){

        try {
            brandService.save(tbBrand);
            return new Result(true, "增加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "服务器正忙");
        }
    }

    /**
     * 根据id查询
     * @return
     */
    @RequestMapping("/findOne.do")
    public TbBrand findOne(Long id){
        TbBrand tbBrand = brandService.findOne(id);
        return tbBrand;
    }


    /**
     * 根据id修改
     * @param tbBrand
     * @return
     */
    @RequestMapping("/update.do")
    public Result update(@RequestBody TbBrand tbBrand){
        try {
            brandService.update(tbBrand);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "服务器正忙");
        }


    }


    /**
     * 批量删除
     * @param ids
     * @return
     */
    @RequestMapping("/remove.do")
    public Result Remove(Long[] ids){
        try{
            brandService.delete(ids);
            return new Result(true,"删除成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"服务器正忙");
        }

    }


    @RequestMapping("/search.do")
    public PageResult<TbBrand> search(@RequestBody TbBrand tbBrand,Integer currPage,Integer pageSize ){

        return brandService.findByPage(tbBrand,currPage,pageSize);
    }





}

File: pinyougou-parent-module/pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController', function ($scope,$controller,brandService) {

    $controller('baseController',{$scope:$scope});


    //分页查询(已代替)
    /*$scope.findPage = function (currPage, pageSize) {
        $http.get('../brand/findByPage.do?currPage=' + currPage + '&pageSize=' + pageSize).success(
            function (data) {
                $scope.list = data.rows;
                $scope.paginationConf.totalItems = data.total;//更新总记录数
            }
        );
    };*/

    //新增/修改品牌
    $scope.save = function () {
        var object = brandService.save($scope.entity);
        if ($scope.entity.id != null) {
            object = brandService.update($scope.entity);
        }

        object.success(
            function (data) {
                if (data.success) {
                    $scope.reloadList();//重新加载
                } else {
                    alert(data.message);
                }
            }
        )
    };

    //修改回显
    $scope.findOne = function (id) {
        brandService.findOne(id).success(
            function (data) {
                $scope.entity = data;
            }
        )
    };

    //批量删除
    $scope.remove = function () {

        if (confirm("确定要删除吗?")) {
            brandService.remove($scope.selectIds).success(
                function (data) {
                    if (data.success) {
                        $scope.reloadList();
                    } else {
                        alert(data.message);
                    }

                }
            )
        }
        //每次删除操作后情况id数组（建议将此操作放入reloadList）
        // $scope.selectIds = [];
    };

    //查询条件
    $scope.searchEntity= {};

    //条件分页查询
    $scope.search = function (currPage,pageSize) {
        brandService.search(currPage,pageSize,$scope.searchEntity).success(
            function (data) {
                $scope.list = data.rows;
                $scope.paginationConf.totalItems = data.total;//更新总记录数
            }
        )
    }


});